### PR TITLE
fix(space-nuxt-base): upgrade app-extension-auth

### PR DESCRIPTION
**Pull request by eunjae-lee**
## What?

This PR upgrads app-extension-auth to v2.

### CODE REVIEW

File: space-plugins/nuxt-base/server/middleware/02.auth.global.ts [hunk 1]
```diff
@@ -17,11 +17,9 @@ export default defineEventHandler(async (event) => {
 	// (Storyfront attaches this query parameter in that case)
 	const queryParams = getQuery(event);
 	if (queryParams['init_oauth'] === 'true') {
-		return await sendRedirect(
-			event,
-			`${appConfig.auth.initOauthFlowUrl}?init_oauth=true`,
-			302,
-		);
+		const sep = appConfig.auth.initOauthFlowUrl.includes('?') ? '&' : '?';
+		const url = `${appConfig.auth.initOauthFlowUrl}${sep}init_oauth=true`;
```
Comment (BibiSebi) [inline space-plugins/nuxt-base/server/middleware/02.auth.global.ts:21]: The `init_oauth` is currently not being used in the app-extension auth right? Should we still forward it?

Comment (eunjae-lee) [reply space-plugins/nuxt-base/server/middleware/02.auth.global.ts:21]: true. let's not!

**Review by BibiSebi — APPROVED** [review]
Let's go :)

File: space-plugins/nuxt-base/server/utils/getAppSession.ts [hunk 2]
```diff
@@ -29,93 +24,25 @@ type AppSessionQuery = {
...
-
-	if (appSession?.spaceId && appSession?.userId) {
-		// When a page is already authenticated on the server side,
-		// and it's rendering a page that includes `useFetch('...', { server: true })`,
-		// Nuxt directly runs the serverless function on the server side
-		// (without an actual HTTP request).
-		//
-		// In that case `event.context.appSession` exists.
-		return convertToAppSessionQuery(appSession?.spaceId, appSession?.userId);
+	if (appSession && appSession.spaceId && appSession.userId) {
```
Comment (BibiSebi) [inline space-plugins/nuxt-base/server/utils/getAppSession.ts:36]: maybe we can even expose isAppSession validation method which we could use here?

**Review by BibiSebi — APPROVED** [review]
Looks very neat.

**Comment by BibiSebi** [comment]
nned to wait for next release as inferSession function is not released yet

**Comment by BibiSebi** [comment]
Everything is updated, but still waiting before we finalise the changes on app-extension-auth before merging this.

File: space-plugins/nuxt-base/server/utils/getAppSession.ts (hunk 2 above)
Comment (eunjae-lee) [reply space-plugins/nuxt-base/server/utils/getAppSession.ts:36]: that's a good point.

**Pull request merged**